pkg/util/indexparamcheck/scann_checker.go

```diff
@@ -0,0 +1,41 @@
+package indexparamcheck
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// scaNNChecker checks if a SCANN index can be built.
+type scaNNChecker struct {
+	ivfBaseChecker
+}
+
+// CheckTrain checks if SCANN index can be built with the specific index parameters.
+func (c *scaNNChecker) CheckTrain(params map[string]string) error {
+	if err := c.ivfBaseChecker.CheckTrain(params); err != nil {
+		return err
+	}
+
+	return c.checkScaNNParams(params)
+}
+
+func (c *scaNNChecker) checkScaNNParams(params map[string]string) error {
```

Comment:
Please add ut for all the corner cases.

Reply:
scann's ut is more likely the subset of ivfpq, I guess it is enough